prov/tcp/src/tcpx_rma.c

```diff
@@ -79,24 +69,18 @@ static void tcpx_rma_read_recv_entry_fill(struct tcpx_xfer_entry *recv_entry,
 					  const struct fi_msg_rma *msg,
 					  uint64_t flags)
 {
-	uint64_t data_len;
+	recv_entry->msg_hdr.hdr.size =
+		htonll(sizeof(recv_entry->msg_hdr) +
+		       ofi_total_iov_len(msg->msg_iov, msg->iov_count));
 
-	data_len = ofi_total_iov_len(msg->msg_iov, msg->iov_count);
+	memcpy(&recv_entry->msg_data.iov[0], &msg->msg_iov[0],
+	       msg->iov_count * sizeof(struct iovec));
```

Comment:
This is just relocated code, but are either of these iovecs placed on the wire?

Reply:
They are not placed on wire. Msg_data.iov is holder for the destination addresses for data in RMA read operation. I am not really using the hdr. I can get rid of it.